driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -2208,16 +2208,35 @@ public void ensurePoolsSizing() {
         }
 
         public PreparedStatement addPrepared(PreparedStatement stmt) {
```

Comment:
> if it's being re-prepared because the resultset metadata has changed.
This is no longer the case. When the resultset metadata has changed, we don't re-prepare and re-add the statement, only mutate it in place.
So the only way `addPrepared` is called twice is if the client explicitly called `session.prepare` twice with the same statement. Whether the resultset metadata changed or not between the two calls is irrelevant I think, this is still an anti-pattern.